server/protocol/ping.js

```diff
@@ -40,6 +40,18 @@ module.exports = function createPingHandler(ringpop) {
             return callback(new Error('need req body with source, changes, and checksum'));
         }
 
+        if (body.hasOwnProperty('app')) {
+            if (body.app !== ringpop.app) {
+                ringpop.logger.warn('Rejected ping from wrong ringpop app', body);
```

Comment:
I think warning is better because if another service is pinging incorrectly, it's not an error for the pinged service.